lib/zone.ts

```diff
@@ -136,11 +136,11 @@
 interface Zone {
   /**
    *
-   * @returns {Zone} The parent Zone.
+   * @returns {?Zone} The parent Zone.
```

Comment:
what is the reasoning for updating all the inline types?

If the externs file is independent of the .ts file,
1) we'll have to maintain two different copies of the same info, and
2) nothing enforces the JSDoc annotations in the .ts file are correct, so we will probably fail to maintain them

angular/tsickle can generate externs from the TS types, so we could just remove the JSDoc annotations if we were using the extern generator.

I know you already did a lot of work here, sorry to have to ask this!